tests/integration/index/query_with_compound_filter_relation_test.go

```diff
@@ -0,0 +1,362 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package index
+
+import (
+	"testing"
+
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+)
+
+func TestIndex_QueryWithIndexOnOneToManyRelationAndFilter_NoData(t *testing.T) {
+	test := testUtils.TestCase{
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+				  type Program {
+					name: String
+					certificationBodyOrg: Organization
+				  }
+
+				  type Organization {
+					name: String @index
+					programs: [Program]
+				  }`,
+			},
+			testUtils.Request{
+				Request: `query {
+					Program(
+						filter: {
+							_and: [
+								{ certificationBodyOrg: { name: { _eq: "Test" } } }
+							]
+						}
+					) {
+						name
+					}
+				}`,
+				Results: []map[string]any{},
+			},
+		},
+	}
+
+	testUtils.ExecuteTestCase(t, test)
+}
+
+func TestIndex_QueryWithIndexOnOneToManyRelationOrFilter_NoData(t *testing.T) {
+	test := testUtils.TestCase{
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+				  type Program {
+					name: String
+					certificationBodyOrg: Organization
+				  }
+
+				  type Organization {
+					name: String @index
+					programs: [Program]
+				  }`,
+			},
+			testUtils.Request{
+				Request: `query {
+					Program(
+						filter: {
+							_and: [
```

Comment:
todo: this is not `_or`

Reply:
https://github.com/sourcenetwork/defradb/pull/2581